chore: ensure mypy passes for package

crwilcox opened the pull request:
related #240

```
mypy google/cloud/datastore
Success: no issues found in 12 source files
```

```
mypy -p google.cloud.datastore --no-incremental
Success: no issues found in 12 source files
```

crwilcox committed:
chore: ensure mypy passes

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

crwilcox committed:
chore: flake8 doesn't like lambdas

crwilcox committed:
chore: merge

tseaver commented on google/cloud/datastore/client.py:
```diff
@@ -36,11 +36,14 @@
     from google.api_core import client_info
 
-    make_datastore_api = None
+    def no_datastore_api(self):
+        return None
+
+    make_datastore_api = no_datastore_api
```
Why isn't this just a function?
```suggestion
    def make_datastore_api(self):
        return None
```

And maybe it should raise an exception, since the only reason we're here is that gRPC is disabled?

crwilcox replied:
I get it, but if you do that you end up fighting with `google/cloud/datastore/client.py:41: error: All conditional function variants must have identical signatures` :\ This seems not terrible?

tseaver reviewed (changes requested):
Shouldn't we be adding a nox session for mypy?

crwilcox committed:
test: add mypy test scenarioa

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

crwilcox committed:
chore: scope mypy to google.cloud.datastore for now

crwilcox committed:
chore: remove api_core ignores, quiet a mypy error by import rename

crwilcox commented:
This can merge, but bug should have a comment added that we want to evaluate mypy against `google` and `tests` (comment in code)

This PR needs to merge to remove a class of errors from generated code https://www.github.com/googleapis/gapic-generator-python/issues/1026

tseaver committed:
fix: drop unneded extra function

Also, have shim 'make_datastore_api' raise an exception, as it only
gets called if gRPC is disabled, or not installed.

tseaver commented:
@googlebot I consent.
